apps/remote_control/lib/lexical/remote_control/api.ex

```diff
@@ -13,6 +13,37 @@ defmodule Lexical.RemoteControl.Api do
 
   require Logger
 
+  @spec maybe_schedule_compile(Project.t(), Lexical.uri()) :: :ok
+  def maybe_schedule_compile(project, triggered_file_uri) do
+    if in_rename_progress?(project) do
+      RemoteControl.call(project, Commands.Rename, :mark_saved, [triggered_file_uri])
+    else
+      schedule_compile(project, false)
+    end
```

Comment:
why do these if statements need to happen here rather than on the remote_control side?